dozer-sql/src/pipeline/utils/record_hashtable_key.rs

```diff
@@ -0,0 +1,32 @@
+use ahash::AHasher;
+use dozer_types::types::Field;
+use std::hash::{Hash, Hasher};
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub enum RecordKey {
+    Accurate(Vec<Field>),
+    Hash(u64),
```

Comment:
Given that we've already calculated hash in the construction of `RecordKey`, the implementation of `ProbabilisticHashTable` feels redundant.

Can we discard the generality of the code and make `HashTable::K` `RecordKey`?

Reply:
You're absolutely right. The abstract HashTable was the first attempt, but when I switched to an abstract Key, it made the special HashTable redundant.

Reply:
Sorry but the new usage of `HashMap` seems no different from `ProbabilisticHashTable`, in terms of hash calculating?

We still calculate hashes twice: first to construct `RecordKey`, second to hash `RecordKey` in `HashMap`.

Reply:
Yes, this is how the old implementation used to work. I merely added an accurate implementation besides the old one. I can make a `Hash` trait implementation for `RecordKey` that feeds a precalculated hash to a custom pass-through `Hasher`. What do you think of this idea?

Reply:
You're right. Let's merge it as is.